src/mbgl/mtl/texture2d.cpp

```diff
@@ -75,126 +70,103 @@ size_t Texture2D::numChannels() const noexcept {
     }
 }
 
-/*
+MTL::PixelFormat Texture2D::getMetalPixelFormat() const noexcept {
+    switch (channelType) {
+        case gfx::TextureChannelDataType::UnsignedByte:
+            switch (pixelFormat) {
+                case gfx::TexturePixelType::RGBA:
+                    return MTL::PixelFormat::PixelFormatRGBA8Unorm;
+                case gfx::TexturePixelType::Alpha:
+                    return MTL::PixelFormat::PixelFormatA8Unorm;
+                default:
+                    assert(false);
+            }
+        case gfx::TextureChannelDataType::HalfFloat:
+            switch (pixelFormat) {
+                case gfx::TexturePixelType::RGBA:
+                    return MTL::PixelFormat::PixelFormatRGBA16Float;
+                case gfx::TexturePixelType::Alpha:
+                    return MTL::PixelFormat::PixelFormatR16Float;
+                default:
+                    assert(false);
+            }
+    }
+}
+
 void Texture2D::createObject() noexcept {
```

Comment:
Do we still need a split concept of 'object' and 'storage' in our Metal implementation?

Reply:
Seems harmless enough.

Reply:
I keep similar with GL to catch errors faster. Now that it works I will refactor it a little bit. 👍